### DISCUSSION TITLE
How to include experimental data of OCP entropic change into the simulation?

### DISCUSSION THREAD

**Discussion by JSHUAIFE** [pybamm-team/PyBaMM#1425] [Q&A] [answered]
Hi, 

I am new to PyBaMM. I try to simulate the thermal effects on Li-Ion battery performance.  From the example cases, I noticed that the OCP entropic change can be inputted by functions. However, I failed to input the OCP entropic change from experimental data. I want to know whether it is possible to read in the experimental data  to set OCP entropic change.  If you have any possible shortcut to do so, please suggest!

**Comment by valentinsulzer** [comment, marked as answer]
I'm not sure exactly what you mean - there is more to thermal simulations than just the OCP entropic change.
Do you just want to change the ambient temperature? Or simulate thermal effects in the cell? For the former, just change the "Ambient temperature [K]" parameter. For the latter, see [this notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/using-model-options_thermal-example.ipynb)

**Comment by JSHUAIFE** [comment]
Hi Tinosulzer,

Thanks for your reply.  To account for the heat generation in a Li-ion cell, it is important to include the entropy change (dU/dT) of electrode during charge or discharge, which contributes to the reversible heat generation.  From the input parameters for Li-Ion battery in PyBaMM, I noticed that the OCP entropic change for electrodes can be input by functions (e.g. pybamm\input\parameters\lithium-ion\positive_electrodes\NMC_UMBL_Mohtat2020\NMC_entropic_change_PeymanMPM.py ). For my case, I have the experimental data of entropic change with SoC. Hence, I want to know how to include the experimental data of entropic change in PyBaMM  simulations.

**Reply by valentinsulzer** [reply]
You can specify a lookup table for the OCP entropic change. For example, see how the "Negative electrode OCP [V]" [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium-ion/negative_electrodes/graphite_Chen2020/parameters.csv) calls the csv file in the same folder as a lookup table

**Reply by JSHUAIFE** [reply]
Hi Tinosulzer, 

Thanks for good suggestion. I tried the latest version of PyBaMM v0.3.0. A lookup table for the OCP entropic change can be specified.
